generator/internal/parser/httprule/http_rule_parser.go

```diff
@@ -0,0 +1,382 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httprule
+
+import (
+	"fmt"
+	"github.com/googleapis/google-cloud-rust/generator/internal/api"
+	"strings"
+)
+
+// The following documentation was copied from https://github.com/googleapis/google-cloud-cpp/blob/4174d656136f4b849c8a3d327237f3a96be3e003/generator/internal/http_annotation_parser.h#L49-L58
+// The result of parsing a `google.api.http` annotation.
+//
+// A `google.api.http` annotation describes how to convert gRPC RPCs to HTTP
+// URLs. The description uses a "path template", showing what portions of the
+// URL path are replaced with values from the gRPC request message.
+//
+// These path templates follow a specific grammar. The grammar is defined by:
+//
+//	Template = "/" Segments [ Verb ] ;
+//	Segments = Segment { "/" Segment } ;
+//	Segment  = "*" | "**" | LITERAL | Variable ;
+//	Variable = "{" FieldPath [ "=" Segments ] "}" ;
+//	FieldPath = IDENT { "." IDENT } ;
+//	Verb     = ":" LITERAL ;
+//
+// The specific notation is not defined, but it seems inspired by
+// [Backus-Naur Form]. In this notation, `{ ... }` allows repetition.
+//
+// The documentation goes on to say:
+//
+//	A variable template must not contain other variables.
+//
+// So the grammar is better defined by:
+//
+//	Template = "/" Segments [ Verb ] ;
+//	Segments = Segment { "/" Segment } ;
+//	Segment  = Variable | PlainSegment;
+//	PlainSegment  = "*" | "**" | LITERAL ;
+//	Variable = "{" FieldPath [ "=" PlainSegments ] "}" ;
+//	PlainSegments = PlainSegment { "/" PlainSegment };
+//	FieldPath = IDENT { "." IDENT } ;
+//	Verb     = ":" LITERAL ;
+//
+// Neither "IDENT" nor "LITERAL" are defined. From context we can infer that
+// IDENT must be a valid proto3 identifier, so matching the regular expression
+// `[A-Za-z][A-Za-z0-9_]*`. Likewise, we can infer that LITERAL must be a path
+// segment in a URL. [RFC 3986] provides a definition for these, which we
+// summarize as:
+//
+// LITERAL     = pchar { pchar }
+// pchar       = unreserved | pct-encoded | sub-delims | ":" | "@"
+// unreserved  = ALPHA | DIGIT | "-" | "." | "_" | "~"
+// pct-encoded = "%" HEXDIG HEXDIG
+// sub-delims  = "!" | "$" | "&" | "'" | "(" | ")"
+//
+//	| "*" | "+" | "," | ";" | "="
+//
+// ALPHA       = [A-Za-z]
+// DIGIT       = [0-9]
+// HEXDIG      = [0-9A-Fa-f]
+//
+// [RFC 3986]: https://datatracker.ietf.org/doc/html/rfc3986#section-3.3
+// [Backus-Naur Form]: https://en.wikipedia.org/wiki/Backus%E2%80%93Naur_form
+
+func Parse(pathTemplate string) (*PathTemplate, error) {
+	return parsePathTemplate(pathTemplate)
+}
+
+// ParseSegments flattens the result of Parse into a slice of api.PathSegment,
+// ignoring variable values and match (* and **) segments.
+// TODO: This function is a temporary shim to allow the existing tests to pass.
+func ParseSegments(pathTemplate string) ([]api.PathSegment, error) {
```

Comment:
This function reproduces the behavior of `parseRawPath(string)` that I removed from `protobuf_annotations.go`. I'd need to understand better how this `[]api.PathSegment` is used during code generation to better map the original data structure (or change the code to use that instead). But for now, the tests are passing, so I think we can leave it like this.